### PR TITLE
A checkbox in Query Editor tab in Preferences for disabling syntax highlighting in case the user wants the min input lag

### PR DESCRIPTION
**Pull request by kirill-c**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Added a checkbox in Query Editor tab in Preferences for disabling syntax highlighting in case the user wants the min input lag

Does this close any currently open issues?
------------------------------------------
I don't think so.

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Devices/Simulators:** 2019 MBP 16"

**macOS Version:** 10.15.5

**Sequel-Ace Version:** 

image

### CODE REVIEW

File: Interfaces/Base.lproj/ConnectionView.xib [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ DQ
                         <rect key="frame" x="0.0" y="0.0" width="580" height="320"/>
                         <clipView key="contentView" drawsBackground="NO" copiesOnScroll="NO" translatesAutoresizingMaskIntoConstraints="NO" id="1Qd-RL-aJ4">
                             <rect key="frame" x="0.0" y="0.0" width="580" height="320"/>
-                            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                            <autoresizingMask key="autoresizingMask"/>
                             <subviews>
                                 <textView editable="NO" drawsBackground="NO" importsGraphics="NO" verticallyResizable="NO" id="5434" customClass="SPAutosizingTextView" customModule="Sequel_Ace" customModuleProvider="target">
```
Comment (Jason-Morcos) [inline Interfaces/Base.lproj/ConnectionView.xib:1626]: Please revert changes to this file

File: Interfaces/Base.lproj/Preferences.xib [hunk 2]
```diff
@@ -1538,8 +1538,19 @@ Warning: If Sequel Ace and the server do not have at least one cipher suite in c
...
+                        <behavior key="behavior" changeContents="YES" doesNotDimImage="YES" lightByContents="YES"/>
+                        <font key="font" metaFont="system"/>
+                    </buttonCell>
+                    <connections>
+                        <binding destination="117" name="value" keyPath="values.CustomQueryEnableSyntaxHighlighting" id="kJw-7M-n52"/>
+                    </connections>
+                </button>
             </subviews>
-            <point key="canvasLocation" x="200" y="1295"/>
+            <point key="canvasLocation" x="200" y="1295.5"/>
```
Comment (Jason-Morcos) [inline Interfaces/Base.lproj/Preferences.xib:1553]: Could you move the checkbox to the right side?

Comment (Jason-Morcos) [reply Interfaces/Base.lproj/Preferences.xib:1553]: Perhaps above the auto-completion checkbox as well, I believe it makes a little more sense there

File: Interfaces/Base.lproj/ConnectionView.xib [hunk 3]
```diff
@@ -1621,7 +1621,7 @@ DQ
...
                                     <rect key="frame" x="0.0" y="0.0" width="580" height="320"/>
```
Comment (Jason-Morcos) [inline Interfaces/Base.lproj/ConnectionView.xib:1627]: Please revert all changes to this file as they seem unrelated to the PR

**Comment by kirill-c** [comment]
Thank you for taking a look, Jason.

I've made the changes you requested.

**Comment by Jason-Morcos** [comment]
Awesome!

One more thing - I just checked it out locally and it appears to work well. The only funky thing is disabling syntax highlighting does _not_ cause the text view to reload _without_ syntax highlighting. In other words, if you first type content in the query editor when highlighting is off, when you turn it on the text gets syntax highlighted magically. However, when disabling syntax highlighting after typing text in the query editor when syntax highlighting is on, the highlighting does _not_ get removed. The last little change would be to try to make that happen - to make disabling syntax highlighting cause the text view to remove any highlighting it had added.

**Comment by kirill-c** [comment]
I thought having left-over highlighting would be better than not having it at all but I agree and implemented a fix to make the checkbox' behavior more consistent

**Comment by kirill-c** [comment]
Thank you Jason

**Comment by Jason-Morcos** [comment]
> I thought having some left-over highlighting would be better than not having it at all but I agree and implemented a fix to make the checkbox' behavior more consistent

We're getting so close! Now the issue for me is it turns my text black when disabling syntax highlighting. This is because I have set a custom text color. Instead of dropping all color, we need to fallback on the user's chosen text color. I believe that's `otherTextColor`

Also, more a style thing, but I like the variable `syntaxHighlightingApplied` versus `syntaxHighlightingRemoved` (inverses meaning). It's more generalizable for future things and makes a little more natural sense (we're only removing highlighting if it's applied)

**Comment by Jason-Morcos** [comment]
One other tiny style thing, the logic to remove syntax highlighting would be good to abstract into it's own method: 
`- (void)removeSyntaxHighlighting`

**Comment by kirill-c** [comment]
Updated

**Comment by kirill-c** [comment]
Thank you!

**Comment by kirill-c** [comment]
May I merge?

**Comment by kirill-c** [comment]
🙂

**Comment by Jason-Morcos** [comment]
Will include this in the 2.1.3 release! Will be included in next beta and there'll probably be a new beta later today or tomorrow

**Comment by kirill-c** [comment]
thanks!
